Branch-SDK/src/main/java/io/branch/referral/Branch.java

```diff
@@ -2257,200 +2234,12 @@ private boolean checkIntentForUnusedBranchLink(Intent intent) {
-            } else {
-                return branchRemoteInterface_.make_restful_post(thisReq_.getPostWithInstrumentationValues(instrumentationExtraData_), thisReq_.getRequestUrl(), thisReq_.getRequestPath(), prefHelper_.getBranchKey());
-            }
-        }
-        
-        @Override
-        protected void onPostExecute(ServerResponse serverResponse) {
-            super.onPostExecute(serverResponse);
-            latch_.countDown();
-
-            if (serverResponse == null || isCancelled()) return;
-
-            try {
-                int status = serverResponse.getStatusCode();
-                hasNetwork_ = true;
-
-                if (serverResponse.getStatusCode() == BranchError.ERR_BRANCH_TRACKING_DISABLED) {
-                    thisReq_.reportTrackingDisabledError();
-                    requestQueue_.remove(thisReq_);
-
-                } else {
-                    //If the request is not succeeded
-                    if (status != 200) {
-                        //If failed request is an initialisation request then mark session not initialised
-                        if (thisReq_ instanceof ServerRequestInitSession) {
-                            setInitState(SESSION_STATE.UNINITIALISED);
-                        }
-                        // On a bad request or in canse of a conflict notify with call back and remove the request.
-                        if (status == 400 || status == 409) {
-                            requestQueue_.remove(thisReq_);
-                            if (thisReq_ instanceof ServerRequestCreateUrl) {
-                                ((ServerRequestCreateUrl) thisReq_).handleDuplicateURLError();
-                            } else {
-                                PrefHelper.LogAlways("Branch API Error: Conflicting resource error code from API");
-                                handleFailure(0, status);
-                            }
-                        }
-                        //On Network error or Branch is down fail all the pending requests in the queue except
-                        //for request which need to be replayed on failure.
-                        else {
-                            hasNetwork_ = false;
-                            //Collect all request from the queue which need to be failed.
-                            ArrayList<ServerRequest> requestToFail = new ArrayList<>();
-                            for (int i = 0; i < requestQueue_.getSize(); i++) {
-                                requestToFail.add(requestQueue_.peekAt(i));
-                            }
-                            //Remove the requests from the request queue first
-                            for (ServerRequest req : requestToFail) {
-                                if (req == null || !req.shouldRetryOnFail()) { // Should remove any nullified request object also from queue
-                                    requestQueue_.remove(req);
```

Comment:
we used to remove ALL (except `v1/event` and `v1/profile`) requests from the queue once we encounter the first failed request whose status code is not 400/409. I disagree with the assumption that all subsequent requests will fail. With the flaky connections on mobile phones being so common, the next request could succeed just fine... To that end, I now only remove the recently completed request from the queue (unless it failed and it's a `v1/event` or `v1/profile` event) but I leave all other requests on the queue.